syntax = "proto3";

package wowlogfetcher;

service LogFetcher {
  
  rpc GetDungeonRanks (DRRequest) returns (DRResponse) {}

  rpc GetRaidRanks (RRRequest) returns (RRResponse) {}

  rpc AddGuildToWathcer(GuildAddRequest) returns (GuildAddResponse) {}

  rpc AddCharacterToWathcer(CharacterAddRequest) returns (CharacterAddResponse) {}

}


message DRRequest {
  string dung = 1;
  string rio_link = 2;
}

message RRRequest {
  string raid = 1;
  optional int32 dfffc = 2;
  string rio_link = 3;
}


message BossResponse{
  repeated int32 rankings = 1;
}

message DRResponse {
  string name = 1;
  repeated int32 rankings = 2;
}

message RRResponse {
  string name = 1;
  map<string, BossResponse> rankings = 2;
}

message GuildAddRequest {
  string rio_guild_link = 1;
  string channel_id = 2;
}

message GuildAddResponse {
  int32 rd_guild_id = 1;
}

message CharacterAddRequest {
  string rio_character_link = 1;
  string channel_id = 2;
}

message CharacterAddResponse {
  int32 rd_char_id = 1;
}

